infor-design/enterprise
# 5091 - Fix visual bugs in editable Datagrid time/date/lookup cells

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR makes a few adjustments to the visual state of Date/Time/Lookup cells:
- There was previously a bug where the focus state could get "stuck" on Date and Time cells after a specific sequence of enter/exit of cell edit mode.  This has been fixed.
- There were occasionally some visual bugs related to the background color of trigger icons/buttons in Time/Date/Lookup cells.  These bugs have been fixed.

**Related github/jira issue (required)**:
Closes #5091 
Closes #5157

**Steps necessary to review your pull request (required)**:

_To test background color fixes:_
- Go to http://localhost:4000/components/datagrid/example-editable.html
- Click on a timepicker/datepicker trigger button inside a cell
- Click outside of the datagrid
- Hover over the row with the icon button that was clicked.  The icon should have no discoloration around the outside of it (no white border against the gray hover state).
- Also test this against Lookup cells on http://localhost:4000/components/datagrid/test-editable-lookup.html
- Also observe that error cells don't have the white discoloration either.

_ To test permanent focus state fix:_
- Go to http://localhost:4000/components/datagrid/example-editable.html
- Click on timepicker/datepicker trigger button inside a cell
- Click on another cell of the datagrid that does not have a timepicker/datepicker icon button
- Click the same timepicker/datepicker icon button
- Click away from the cell with the timepicker/datepicker icon button.  The focus state should disappear as expected.

**Included in this Pull Request**:
